src/3DFactory.R

```diff
@@ -0,0 +1,138 @@
+# LT 17/05/2022
+
+# prototype: 3D Factory AND Adjudicator are in the same source file to avoid
+# implementation of the interface between them
+
+# 3D Factory component
+
+# the 3D Factory generates a list of 3D regions and calculates their intolerance score
+
+# First prototype
+# creates 3D regions of variable length using upper bound of CI
+# each region is built by growing a sphere centered on the carbon-alpha of
+# a specific residue until the whole protein is included
+# a criteria (based on the upper bound) is used to decide the length of the region
+
+# file names and parameters are hard-coded for this prototype
+
+source("constraint_utils.R")
+
+## calculate upper bound of the o/e confidence interval
+get_oe_ci_up <- function(observed, expected, alpha = 0.05) {
+  return(qchisq((1 - alpha / 2), 2 * (observed + 1)) / 2 / expected)
+}
+
+## calculate intolerance metric of a region
+get_region_metric <- function(i_region, alpha, regions, oe_data) {
+  # indices of the residues belonging to this region
+  this_region <- regions[seq_len(i_region)]

```

Comment:
What do you mean by ‘this_region’? I’d change this to be more descriptive

Reply:
Agreed. Changed to `region_indices`

Reply:
Or potentially just `region < regions[ ... ]` showing it's a subset? `region_indices` could replace `i_region`